src/docs/development/testing.mdx

```diff
@@ -38,24 +80,90 @@ yarn test
+
+One way to combat this in your own applications is to use dedicated attributes for testing (e.g., `test-id` or `dev-id`). By adding this test-specific attribute to the element, identifying the correct underlying component should be safer. Because this attribute is test-specific and the application logic does not use it, it will be less likely to change throughout the product lifecycle. Combining this approach with other selectors is a common and reliable approach for element identification.
+
+```html
+<button test-id="login-button" class="default-button">Login</button>
+```
+
+Selector to use in a test
+
+```
+WebElement click = driver.findElement(By.id("login-button"));
+```
+
+## Testing Frameworks
+
+There are many tools available for configuring automated E2E tests. Some of the more popular tools include:
+
+-   Selenium
+-   Cypress
+-   WebDriverIO
+
+### Selenium
+
+Selenium is a popular automation testing suite which can be used to automate the desktop and mobile web browser interactions. You can perform automation testing by writing code in any of your preferred language supported by Selenium.
+
+Selenium test suite has several test frameworks available and each one can be custmized to your project needs.
+
+For more information see [browserstack](https://www.browserstack.com/guide/selenium-framework) selenium guide or [selenium on github](https://github.com/SeleniumHQ/selenium) documentation.
+
+For selenium tutorial series see [software testing help](https://www.softwaretestinghelp.com/selenium-tutorial-1/)
 
 ### Cypress
 
-Cypress is an open source testing framework that boasts support for any modern JavaScript framework. It works well for E2E tests in both Angular and React applications and is easier to use than some of the more traditional tools, such as Selenium.
+Cypress is a popular open source testing framework that boasts support for any modern JavaScript framework. It works well for E2E tests in both Angular and React applications and is easier to use than some of the more traditional tools, such as Selenium.
 
 For a more detailed write-up on Cypress, check out the official [Cypress Documentation](https://docs.cypress.io/guides/getting-started/installing-cypress.html#System-requirements).
+
+### WebDriverIO
+
+WebdriverIO is a popular Javascript based test automation framework built over nodeJs. It is an open-source project developed for the automation testing community. WebdriverIO is extendible, compatible and feature-rich.
+
+For more information see [webdriver.io](https://webdriver.io/) and [webdriver.io](https://webdriver.io/docs/gettingstarted.html) getting started.
+
+<Divider />
+
+# Mobile Device Testing
+
+Coming soon...
```

Comment:
Should we add something here to talk about the different options that are available (at a high level)? Or just leave this our for now until later?